### ISSUE TITLE
Using `setAttribute` for rotation data can lead to inconsistent state between clients.

### ISSUE THREAD

**Issue by johnshaughnessy** [networked-aframe/networked-aframe#134]
Suppose I do the following:
- Add a schema for my object, `foo`, whose rotation I want sent over the network:
```
  NAF.schemas.add({
    template: "#foo",
    components: [ "rotation"]
  });
```
- Create a `foo`:
```
   const foo = document.createEntity("a-entity");
   foo.setAttribute("networked", { template: "#foo" });
   scene.appendChild(foo);
```
- Change its rotation via its object3D:
```
  foo.object3D.quaternion.copy(otherQuaternion);
```
- On the sender's side, `networked` will `gatherComponentData` to get the `rotation` from the `object3D`. Note that `foo` does not have an aframe `rotation` component:
https://github.com/aframevr/aframe/commit/2341040d07518cc3f35f704d570ae8c36eba824b#diff-1a1e0c9c4d54c218dff1af318ae8fe5cR726

- On the receiver's side, `networked` will call `setAttribute` and thus create an aframe `rotation` component. 

- The `rotation` component on the receiver's side will change the euler order of receiver's copy of `foo`:
https://github.com/aframevr/aframe/commit/801bbfcf86762b2e90cbc1838ed62d4c43898817#diff-7431e45a2a424db19d55431a61218c27R25

- Now, the sender's `foo` has its euler order set to the default order, `XYZ`, and the receiver's `foo` has its euler order set to `YXZ`. Thus, rotations applied by the receiver are wrong.

**Comment by wmurphyrd** [comment]
I encountered this same issue today. In my case, setting the orientation via `object3D.rotation` instead of the quaternion resolved it, but that's not always an option (e.g. if the updates are coming from `aframe-physics-system`)

**Comment by Huggoz** [comment]
I believe I am having the same issue. please help :) I've been spending hours trying to figure out why when entering vr mode rotation doesn't update for other users. Teleport works fine, updating position works too. Here is my project [https://glitch.com/~forevrmall](url)

thank you so much!

**Comment by haydenjameslee** [comment]
@johnshaughnessy do you think the solution here is for NAF to directly change the position, rotation and scale of the object3D?

**Comment by kylebakerio** [comment]
I'm wondering if I am still running into this same issue; is that possible? Was this changed, and just not closed, or is it still a valid issue?

**Comment by Gllloria** [comment]
@wmurphyrd Can you please share your solution? 
I'm still running into the same issue, for some objects euler order 'YXZ' is correctly working, for some euler order 'XYZ'

**Comment by vincentfretin** [comment]
Yes this can still be an issue I think if you update directly object3D.rotation or object3D.quaternion (like maybe it's done by the physics engine) so without using the aframe rotation component that set the "YXZ" order. If you put the rotation component in your networked template, this should resolve the issue I think, it will force the "YXZ" order before the rotation data are set by the networked entity. I didn't test that though.
